### PR TITLE
Refactor most of badges again

### PR DESCRIPTION
**Pull request by EXtremeExploit**
* Format badges/index.php
* Fix sorting placeholder being `Username`.
* Order badges functions
* use correct variable declaration keywords based on the context its being used
* Fix translations not loading sometimes in badges
* Optimize viewtype change, it really is fast now, try it i swear to god i love it

This is second part of badges "rewrite". if there is a third part it will probably be some speed stuff or smth, theres not much to do now, this may be last or not

### CODE REVIEW

File: badges/index.php [hunk 1]
```diff
@@ -59,21 +59,30 @@
...
-            notification_system();
+    notification_system();
     fontawesome();
     ?>
 </head>
 
 <body>
+
+    <script>
+        const BADGES =
```
Comment (Tanza3D) [inline badges/index.php:79]: please don't do this. we rewrote the entire website specifically to stop printing data directly onto the site.
what is most important, is time from click to first draw, this probably increases that quite a bit. i don't care how long badges take to pop up, as long as the navbar and such is there sooner

**Review by Tanza3D — CHANGES_REQUESTED** [review]
s

File: badges/index.php (hunk 1 above)
Comment (EXtremeExploit) [reply badges/index.php]: donedid

File: badges/index.php [hunk 2]
```diff
@@ -166,15 +166,4 @@
...
-
-<script>
-    function testDialog() {
-        openDialog("Test Title", "Test Header", "Test Message", "Button 1", function() {
-            document.getElementById("testdialog_text").innerHTML = "you clicked button 1!";
-        }, "Button 2", function() {
-            document.getElementById("testdialog_text").innerHTML = "you clicked button 2!";
-        });
-    }
-</script>
```
Comment (Tanza3D) [inline badges/index.php:178]: lmao good point

File: badges/js/functions.js [hunk 3]
```diff
@@ -1,33 +1,30 @@
-var data;
+let BADGES = [];
```
Comment (Tanza3D) [inline badges/js/functions.js:1]: ...why capital? we don't use this anywhere else

**Review by Tanza3D — COMMENTED** [review]
few things

File: badges/js/functions.js (hunk 3 above)
Comment (EXtremeExploit) [reply badges/js/functions.js:1]: true true, i thought it being capital cause its only setted once at the start and then its just being read, like a semi-constant

Comment (Tanza3D) [reply badges/js/functions.js:1]: it kinda makes sense, i understand why, but we literally don't use this anywhere else so probably best not to lol

**Review by Tanza3D — APPROVED** [review]
yay

**Pull request merged**